docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,383 @@
+                "breakfast-included": false,
+                "city": "Berlin",
+                "free-wifi": false,
+                "star-rating": 1,
+                "swimming-pool": false
+            },
+        ]
+    }
+
+The above json file contains data about restaurants and hotels.
+The json structure should contain a key for every object type, i.e. "restaurant" and "hotel".
+Every object type maps to a list of objects.
+
+Once the data are defined in a json file, called, for example, ``data.json``, you can create your
+``InMemoryKnowledgeBase``:
+
+.. code-block:: python
+
+    knowledge_base = InMemoryKnowledgeBase.load("data.json")
+
+Customize your InMemoryKnowledgeBase
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The class ``InMemoryKnowledgeBase`` inherits ``KnowledgeBase``.
+You can customize your ``InMemoryKnowledgeBase`` by overwriting the following functions:
+
+- ``get_key_attribute_of_object``: To keep track of what object the user was talking about last, we store the value
+  of the key attribute in a specific slot. Every object should have a key attribute that is unique, i.e.
+  similar to the primary key in a relation database. Per default the name of the key attribute for every object type
+  is set to "id". You can overwrite the name of the key attribute for a specific object type by calling
+  ``set_key_attribute_of_object()``.
+- ``get_representation_function_of_object``: This methods returns a lambda function that maps an object from the
+  knowledge base to a string representation. This function is used whenever an object is outputted to the user.
+  Per default the lambda function is set to ``lambda obj: obj["name"]``. So, it returns the value of the attribute
+  "name" of the object. If your object does not have an attribute "name", or the "name" of an object might be
+  ambiguous, you should set a new lambda function for that object type by calling
+  ``set_representation_function_of_object()``.
+- ``set_ordinal_mention_mapping``: The ordinal mention mapping is needed to resolve an ordinal mention to an object
+  in a list. For example, if the bot listed a few restaurants in Berlin, and the user then asked "Does the second one
+  have outside seating?", you need to resolve "second one" to the correct object the bot listed before. Per
+  default the ordinal mention mapping looks like this:
+
+  .. code-block:: python
+
+      {
+          "1": lambda l: l[0],
+          "2": lambda l: l[1],
+          "3": lambda l: l[2],
+          "4": lambda l: l[3],
+          "5": lambda l: l[4],
+          "6": lambda l: l[5],
+          "7": lambda l: l[6],
+          "8": lambda l: l[7],
+          "9": lambda l: l[8],
+          "10": lambda l: l[9],
+          "ANY": lambda l: random.choice(list),
+          "LAST": lambda l: l[-1],
+      }
+
+  You can overwrite it by calling the function ``set_ordinal_mention_mapping``.
+  More on the usage can be found in section :ref:`resolve_mentions`.
```

Comment:
Avoid passive voice like "More on the usage can be found". Instead say:
"If you want to learn more about X, go to ..."